packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart

```diff
@@ -109,13 +109,14 @@ class FuchsiaPM {
     if (fuchsiaArtifacts.pm == null) {
       throwToolExit('Fuchsia pm tool not found');
     }
+    host = host.replaceAll('%', '%25');
     final List<String> command = <String>[
       fuchsiaArtifacts.pm.path,
       'serve',
       '-repo',
       repoPath,
       '-l',
-      '$host:$port',
+      '[$host]:$port',
```

Comment:
Is this guaranteed to be ipv6?

Reply:
Hmm it might need that check. Good catch

Reply:
Done! PTAL